engine/runtime-language-arrow/src/main/java/org/enso/interpreter/arrow/runtime/ArrowFixedSizeArrayBuilder.java

```diff
@@ -59,27 +69,21 @@ Object getMembers(boolean includeInternal) throws UnsupportedMessageException {
 
   @ExportMessage
   Object invokeMember(
-      String name,
-      Object[] args,
-      @Cached.Shared("interop") @CachedLibrary(limit = "1") InteropLibrary iop)
+      String name, Object[] args, @Cached(value = "build()") WriteToBuilderNode writeToBuilderNode)
```

Comment:
The `WriteToBuilderNode` is going to be allocated even for `build` operation while it is only needed for `append`. Try:
```java
@Cached(value = "buildWriterOrNull(name)")
```
and add
```java
static WriteToBuilderNode buildWriterOrNull(String op) {
  return "append".equals(op) ? WriterToBuilderNode.build() : null;
}
```
or `WriterToBuilderNode.getUncached()` instead of `null`.